hooks/pre/31-upgrade-puppet.rb

```diff
@@ -25,9 +25,9 @@ def remove_puppet_port_httpd
 
 def copy_data
   success = []
-  success << Kafo::Helpers.execute('cp -rfp /etc/puppet/environments/* /etc/puppetlabs/code/environments') if File.directory?('/etc/puppet/environments')
-  success << Kafo::Helpers.execute('mv /var/lib/puppet/ssl /etc/puppetlabs/puppet && ln -s /etc/puppetlabs/puppet/ssl /var/lib/puppet/ssl') if File.directory?('/var/lib/puppet/ssl')
-  success << Kafo::Helpers.execute('mv /var/lib/puppet/foreman_cache_data /opt/puppetlabs/puppet/cache/') if File.directory?('/var/lib/puppet/foreman_cache_data')
+  success << Kafo::Helpers.execute('cp -rfp /etc/puppet/environments /etc/puppetlabs/code/environments') if File.directory?('/etc/puppet/environments')
+  success << Kafo::Helpers.execute('cp -rfp /var/lib/puppet/ssl /etc/puppetlabs/puppet && ln -s /etc/puppetlabs/puppet/ssl /var/lib/puppet/ssl') if File.directory?('/var/lib/puppet/ssl')
```

Comment:
The symbol link doesn't work right, you end up creatingf /var/lib/puppet/ssl/ssl here.  We can remove the symbolic link if we're going to copy the data.  Or remove /var/lib/puppet/ssl after the copy